core/templates/container_converters.h

```diff
@@ -0,0 +1,103 @@
+/**************************************************************************/
+/*  container_converters.h                                                */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef CONTAINER_CONVERTERS_H
+#define CONTAINER_CONVERTERS_H
+
+#include "core/variant/typed_array.h"
+
+namespace converters {
+
+template <typename T>
+TypedArray<T> vector_to_typed_array(const Vector<T> &p_vector) {
+	TypedArray<T> ret;
+	ret.resize(p_vector.size());
+
+	int i = 0;
+	for (const T &E : p_vector) {
+		ret[i] = E;
+		i++;
```

Comment:
```suggestion
	TypedArray<T>::Iterator itr = ret.begin();
	for (const T &E : p_vector) {
		*itr = E;
		itr++;
```
Should be able to use iterators here for convenience, should be a little more performant (though not necessarily that critical here)

Reply:
Had to use `++itr`, otherwise it wouldn't compile for some reason.

Reply:
Oh yes they only use pre-increment, all the iterators do, probably to avoid the overhead, will look at adding post-increment/decrement to all the iterators as it'd be useful for cases like `*itr++`